File: src/main/java/io/eontimer/service/factory/timer/EntralinkTimerFactory.java
package io.eontimer.service.factory.timer;

import java.time.Duration;
import java.util.Arrays;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import io.eontimer.util.TimeUtil;

@Service
public class EntralinkTimerFactory {

	@Autowired
	private DelayTimerFactory delayTimer;

	public List<Duration> createStages(long targetSecond, long targetDelay, long calibration, long entralinkCalibration) {
		final List<Duration> stages = delayTimer.createStages(targetSecond, targetDelay, calibration);
		return Arrays.asList(stage1(stages), stage2(stages, entralinkCalibration));
	}

	public long calibrate(long targetDelay, long delayHit) {
		return delayTimer.calibrate(targetDelay, delayHit);
	}

	private Duration stage1(List<Duration> stages) {
		return stages.get(0).plus(TimeUtil.milliseconds(250L));
	}

	private Duration stage2(List<Duration> stages, long entralinkCalibration) {
		return stages.get(1).minus(TimeUtil.milliseconds(entralinkCalibration));
	}

}

File: src/service/factory/EntralinkTimerFactory.js
import delayTimerFactory from "./DelayTimerFactory";

const createStages = (targetSecond = 0, targetDelay = 0, calibration = 0, entralinkCalibration = 0, handheld) => {
  const stages = delayTimerFactory.createStages(targetSecond, targetDelay, calibration, handheld);
  return [stage1(stages), stage2(stages, entralinkCalibration)];
};

const calibrate = (targetDelay = 0, delayHit = 0, handheld) => {
  return delayTimerFactory.calibrate(targetDelay, delayHit, handheld);
};

const stage1 = (stages) => {
  return stages[0] + 250;
};

const stage2 = (stages, entralinkCalibration) => {
  return stages[1] - entralinkCalibration;
};

export default {
  createStages,
  calibrate,
};
